lib/mutant/ast/regexp.rb

```diff
@@ -0,0 +1,41 @@
+module Mutant
+  module AST
+    # Regexp source mapper
+    module Regexp
+      UNSUPPORTED_EXPRESSION_TYPE = :conditional
+      private_constant(*constants(false))
+
+      # Check if expression is supported by mapper
+      #
+      # @param expression [Regexp::Expression]
+      #
+      # @return [Boolean]
+      def self.supported?(expression)
+        return if expression.type.equal?(UNSUPPORTED_EXPRESSION_TYPE)
```

Comment:
I might prefer specifying the recursive call site in a test as well. I'll have to play with this rewrite but I think it behaves a bit differently. Writing `return false` for both guards seems more clear to me though